airflow/utils/log/file_task_handler.py

```diff
@@ -317,18 +316,24 @@ def _read(
                 executor_messages, executor_logs = response
             if executor_messages:
                 messages_list.extend(messages_list)
-        if ti.state in (TaskInstanceState.RUNNING, TaskInstanceState.DEFERRED) and not executor_messages:
-            served_messages, served_logs = self._read_from_logs_server(ti, worker_log_rel_path)
-            messages_list.extend(served_messages)
         if not (remote_logs and ti.state not in State.unfinished):
```

Comment:
As discussed on slack, and also mentioned earlier by @zachliu - those conditions are difficult to wrap your head around them  - wrapping them in method names after their meaning ('if _local_or_finished():'` for example) migh help with reviewability :)

Reply:
I agree with you that we can improve readability of these conditions... _however_, with recent discussions fresh in mind, after taking a few steps in that direction (trying to refactor) i realized.... probably best to defer such a refactor (renaming and consolidating logic for the conditionals) for a minor release not the patch release!  😛

the only conditional added here is 
```
ti.state not in State.unfinished and not (local_logs or remote_logs)
```
and this one is pretty plain

it's added as an `elif` block under the moved `if`
WDYT

Reply:
Yeah . for now cool.